config.mk

```diff
@@ -8,9 +8,15 @@ MANPREFIX = ${PREFIX}/share/man
 
 # flags
 CPPFLAGS += -D_DEFAULT_SOURCE -DVERSION=\"${VERSION}\"
-CFLAGS   += -g --std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os ${STATIC} ${CPPFLAGS}
+CFLAGS   += --std=gnu99 -pedantic -Wall -Wno-overlength-strings ${STATIC} ${CPPFLAGS}
 LDFLAGS  += -lm -L${X11LIB}
 
+ifeq (Release, ${BUILD_TYPE})
+CFLAGS += -Os
+else
+CFLAGS += -O0 -g
```

Comment:
I'm not sure about building with debug by default, I'd prefer it if `debug` was actually a separate target since with the way you implemented it now every downstream user will have to add `BUILD_TYPE=Release` before building.

Reply:
Ok, let's explain my thoughts, why I chose this:

```
debug: CFLAGS += -O0 -g
release: CFLAGS += -Os
```
- `make debug` does nothing
- `make debug release all` does shit
- `make debug all` is the right

```
debug: CFLAGS += -O0 -g
debug: all
```

- `make debug` perfect
- You can't build separate targets as `debug`

---

Using a variable is the best thing, as you can only build a single type (and a variable can hold only a single value).

---

> Since now we've been including the debug symbols in the release build (-g) and noone has complained so far. Any specific reason for removing them?

Well, debug symbols are nothing to go into release.

---

every downstream user will have to add BUILD_TYPE=Release before building.
Also if noone has complained, I complain:

```
96K     dunst
320K    dunst.dbg
```

> every downstream user will have to add BUILD_TYPE=Release before building.

Adding a variable to make something work better (it still works without), I consider this a minor change.

I see something else much more worth: Someone opens an issue, saying dunst segfaults regularly. Then you just have to advise the person to build it from the main repo and trigger the condition.

Reply:
> Ok, let's explain my thoughts, why I chose this: [...]

I tried doing something like

```
debug: CFLAGS += -O0

ifeq (debug, $(MAKECMDGOALS))
debug: all
endif
```

seems to work but there's still a warning about `debug` doing nothing in the output if `make debug dunstify` for example is used. What do you think?

But I do see the reasoning behind choosing a variable.

---

> Well, debug symbols are nothing to go into release.
every downstream user will have to add BUILD_TYPE=Release before building.
Also if noone has complained, I complain:

My argument for including the debug symbols in is that this is something to be managed by the distribution, not upstream.

Since I maintain packages in Debian I'll give an example from there: Upstream projects are expected (or the maintainer has to make sure) that they include the full debug symbols and then debhelper strips them from the binary and places them in a separate package as `$(PACKAGENAME)-dbgsym`.

I'd expect that those bypassing the distributions and installing manually either are experienced enough to know to disable debug symbols by themselves, or, like you, will need them sometime.

Reply:
> there's still a warning about debug doing nothing in the output if make debug dunstify

Before introducing new warnings in new code, I think we should go on and find a better solution first.

> My argument for including the debug symbols [... saving the full quote]

Your argumentation makes perfectly sense. I'll change this.

Reply:
> I tried doing something like
> ```
> debug: CFLAGS += -O0
> 
> ifeq (debug, $(MAKECMDGOALS))
> debug: all
> endif
> ```
> seems to work

`dmake debug dunstify` doesn't work. the additional `CFLAGS` are only passed to debug target and its dependencies.

There is also no possibility to define a void target:
```
debug: CFLAGS += -O0

ifeq (debug, $(MAKECMDGOALS))
debug: all
else
debug:
    @true
endif
```

This wouldn't work either.

---

Conclusion: We introduce a debug target, which is only useable as standalone.

Reply:
> Conclusion: We introduce a debug target, which is only useable as standalone.

I pushed this now.